### PR TITLE
[WASMFS] Fix mode on file and dir creation

### PR DESCRIPTION
**Pull request by ethanalee**
Relevant Issue: #15041 

- Add relevant `S_IFREG`, `S_IFDIR`, etc. bits onto file mode.
- Update relevant tests to verify after file and directory creation.

### CODE REVIEW

**Review by tlively — COMMENTED** [review]
Instead of storing the file type in the `kind` field as well as in the mode, can we query `kind` in `stat` to set these bits appropriately in the result?

**Review by tlively — APPROVED** [review]
It would be good to check for the VTX permissions in the tests as well, and to expand the commit message to describe what these permissions are for and point out that they aren't used yet.

File: system/lib/wasmfs/syscalls.cpp [hunk 1]
```diff
@@ -329,6 +335,7 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
     // If O_DIRECTORY is also specified, still create a regular file:
     // https://man7.org/linux/man-pages/man2/open.2.html#BUGS
     if (flags & O_CREAT) {
+      mode &= S_IALLUGO;
```
Comment (kripken) [inline system/lib/wasmfs/syscalls.cpp:338]: Why is the default to give permission to the user, group, and other? I'd expect just user, naively, but probably I'm missing something.

This does seem harmless either way, given we are a single-user environment at least for now. If it's just for now, then a comment at least would be good.

File: system/lib/wasmfs/syscalls.cpp [hunk 2]
```diff
@@ -388,6 +395,7 @@ long __syscall_mkdir(long path, long mode) {
   if (curr) {
     return -EEXIST;
   } else {
+    mode &= S_IRWXUGO | S_ISVTX;
```
Comment (kripken) [inline system/lib/wasmfs/syscalls.cpp:398]: Same questions as above, plus, I think the last flag is the sticky bit - why set that?

File: system/lib/wasmfs/syscalls.cpp (hunk 1 above)
Comment (ethanalee) [reply system/lib/wasmfs/syscalls.cpp:338]: A comment sounds good 👍

File: system/lib/wasmfs/syscalls.cpp (hunk 2 above)
Comment (ethanalee) [reply system/lib/wasmfs/syscalls.cpp:398]: Added comment!

File: tests/wasmfs/wasmfs_mkdir.c [hunk 3]
```diff
@@ -30,6 +30,9 @@ int main() {
   fstat(fdStat, &directory);
 
   assert((directory.st_mode & S_IFMT) == S_IFDIR);
+  assert((directory.st_mode & S_ISVTX) == 0);
+  directory.st_mode &= S_ISVTX;
+  assert((directory.st_mode & S_IRWXUGO) == 0);
```
Comment (tlively) [inline tests/wasmfs/wasmfs_mkdir.c:35]: These two lines aren't testing the behavior of the syscall, so I don't think they're necessary. Same below.

**Review by tlively — CHANGES_REQUESTED** [review]

**Review by kripken — APPROVED** [review]

**Pull request merged**